### PR TITLE
[TASK] Rewrite server-side validators in $asyncValidators (RT-3217)

### PR DESCRIPTION
**Pull request by rht**
Side effect: simplify rpSpinner

### CODE REVIEW

File: src/js/directives/directives.js [hunk 1]
```diff
@@ -321,21 +321,22 @@ module.directive('rpNoPropagate', [function() {
 module.directive('rpSpinner', [function() {
   return {
     restrict: 'A',
-    link: function(scope, element, attr) {
+    require: '^form',
+    link: function(scope, element, attr, ctrl) {
       var spinner = null;
-      attr.$observe('rpSpinner', function(value) {
+      scope.$watch(ctrl.$name+'.'+element.attr('name')+'.$pending', function(pending) {
```
Comment (jublo) [inline src/js/directives/directives.js]: Maybe we should have some spacing between the operators and the strings here.

File: src/js/directives/validators.js [hunk 2]
```diff
@@ -256,76 +221,57 @@ module.directive('rpDest', ['$timeout', '$parse', 'rpFederation', function ($tim
...
-
-        if (timeoutPromise) $timeout.cancel(timeoutPromise);
+      ctrl.$asyncValidators.rpAvailableName = function(value) {
+        scope.usernameInvalidReason = '';
+        scope.usernameReservedFor = '';
 
         if (!value) {
           // No name entered, show nothing, do nothing
-          getterInvalidReason.assign(scope,false);
+          return $q.when(true)
```
Comment (jublo) [inline src/js/directives/validators.js]: Does it look better with `;`?

File: src/js/directives/directives.js (hunk 1 above)
Comment (rht) [inline src/js/directives/directives.js]: Sorry, this was pep8 habit (no spacing between operators that are part of function arguments).

**Comment by mrajvanshy** [comment]
@rht needs to be rebased

**Comment by vhpoet** [comment]
LGTM

**Comment by mrajvanshy** [comment]
LGTM

**Pull request merged**